### PR TITLE
Stop CI Jobs on Forks

### PR DESCRIPTION
**Pull request by jzheaux**
Closes gh-9701

### CODE REVIEW

File: .github/workflows/continuous-integration-workflow.yml [hunk 1]
```diff
@@ -19,9 +19,10 @@ env:
   ARTIFACTORY_PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
 
 jobs:
-  initiate_error_tracking:
-    name: Initiate job-level error tracking
+  parent:
```
Comment (rwinch) [inline .github/workflows/continuous-integration-workflow.yml:22]: Can you make this more descriptive? Perhaps, `prerequisites`?

File: .github/workflows/continuous-integration-workflow.yml [hunk 2]
```diff
@@ -19,9 +19,10 @@ env:
...
+    name: Pre-requisites for building
     runs-on: ubuntu-latest
+    if: github.repository == 'spring-projects/spring-security'
```
Comment (rwinch) [inline .github/workflows/continuous-integration-workflow.yml:25]: Is there a way for this to be done through the environment so users can activate checks in their fork without changing the source?

**Review by rwinch — CHANGES_REQUESTED** [review]
Thanks for the PR! I've left feedback inline

File: .github/workflows/continuous-integration-workflow.yml (hunk 2 above)
Comment (jzheaux) [reply .github/workflows/continuous-integration-workflow.yml:25]: Hypothetically, `prerequisites` could emit some kind of job outcome that the other tasks could check for. For example, they could possibly check for the existence of the `job-initiate-error-tracking.txt` file.

**Review by rwinch — APPROVED** [review]

**Pull request merged**